syntax = "proto3";

package grpcServer;

option go_package = "github.com/lessbutter/alloff-api/api/grpcServer";

service AlloffCategory {
    rpc ListAlloffCategory(ListAlloffCategoryRequest) returns (ListAlloffCategoryResponse);
}

message AlloffCategoryMessage {
    string category_id = 1;
    string name = 2;
    string keyname = 3;
    int32 level = 4;
    string parent_id = 5;
    string img_url = 6;
}


message ListAlloffCategoryRequest {
    string parent_id = 1;
}

message ListAlloffCategoryResponse {
    repeated AlloffCategoryMessage categories = 1;
}